extmod/modussl_axtls.c

```diff
@@ -181,14 +240,43 @@ STATIC mp_uint_t ussl_socket_write(mp_obj_t o_in, const void *buf, mp_uint_t siz
         return MP_STREAM_ERROR;
     }
 
-    mp_int_t r = ssl_write(o->ssl_sock, buf, size);
+    mp_int_t r;
+eagain:
+    r = ssl_write(o->ssl_sock, buf, size);
+    if (r == SSL_OK) {
+        // see comment in read method
+        if (o->blocking) {
+            goto eagain;
+        } else {
+            r = SSL_EAGAIN;
```

Comment:
Is this always ok? I mean if ssl_write wrote all size bytes why does the error has to be EGAIN?

Reply:
So `r == SSL_OK` is really `r == 0` (I'm changing it), which means that nothing got written. If the socket is in blocking mode, we need to loop until at least one byte gets written or an error occurs. In non-blocking mode EAGAIN is the correct return value.

Reply:
> So r == SSL_OK is really r == 0

Ok now it makes sense.